src/rustfmt_diff.rs

```diff
@@ -37,6 +37,43 @@ impl Mismatch {
     }
 }
 
+// This struct handles writing output to stdout and abstracts away the logic
+// of printing in color, if it's possible in the executing environment.
+pub struct OutputWriter {
+    terminal: Option<Box<term::Terminal<Output = io::Stdout>>>,
+}
+
+impl OutputWriter {
+    // Create a new OutputWriter instance based on the caller's preference
+    // for colorized output and the capabilities of the terminal.
+    pub fn new(color: Color) -> Self {
+        if let Some(t) = term::stdout() {
+            if use_colored_tty(color) && t.supports_color() {
+                return OutputWriter { terminal: Some(t) };
+            }
+        }
+        OutputWriter { terminal: None }
+    }
+
+    // Write output in the optionally specified color. The output is written
+    // in the specified color if this OutputWriter instance contains a
+    // Terminal in its `terminal` field.
+    pub fn writeln(&mut self, msg: &str, color: Option<term::color::Color>) {
+        match &mut self.terminal {
+            Some(ref mut t) => {
+                if color.is_some() {
```

Comment:
You can write `if let Some(color) = color {` rather than calling `unwrap()` below.

Reply:
😊  Thanks for catching that.

Reply:
Since it was such a small change, I rewrote history instead of adding a new commit so you can see it easily. You can see the change [here](https://github.com/rust-lang-nursery/rustfmt/pull/2353/commits/9f5f9d2250c03dc915ab949c59dbdbffead5b72b#diff-5b570233633aedc882d38f78b63fa1d5R64).

Thanks again!